tests/test_url.py

```diff
@@ -767,6 +767,101 @@ def test_div_with_dots():
     assert url.raw_path == "/path/to"
 
 
+# joinpath
+
+
+def test_joinpath_root():
+    url = URL("http://example.com")
+    assert str(url.joinpath("path", "to")) == "http://example.com/path/to"
+
+
+def test_joinpath_root_with_slash():
+    url = URL("http://example.com/")
+    assert str(url.joinpath("path", "to")) == "http://example.com/path/to"
+
+
+def test_joinpath():
+    url = URL("http://example.com/path")
+    assert str(url.joinpath("to")) == "http://example.com/path/to"
+
+
+def test_joinpath_with_slash():
+    url = URL("http://example.com/path/")
+    assert str(url.joinpath("to")) == "http://example.com/path/to"
+
+
+def test_joinpath_path_starting_from_slash_is_forbidden():
+    url = URL("http://example.com/path/")
+    with pytest.raises(ValueError):
```

Comment:
It is strongly recommended to make use of the `match` argument when testing with `pytest.raises()` and `pytest.warns()`. Otherwise, this has a chance of causing false positives when it's unintended.